### ISSUE TITLE
MS-SQL tables with triggers are not supported

### ISSUE THREAD

**Issue by paulboudewijn** [directus/directus#9704]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

Content in MS SQL tables with triggers on them cannot be edited via Directus. 
According to the error this is caused by the the use of the OUTPUT clause (by the Directus API), which doesn't work on tables with triggers defined on it. It seems to me that this is caused by the logic trying to get the PK after inserting an item.

https://stackoverflow.com/questions/13198476/cannot-use-update-with-output-clause-when-a-trigger-is-on-the-table

### To Reproduce

1. Create a new table, through Directus or manually in the database.
2. Manually add a trigger to this table.
3. Try to insert a new item to the corresponding collection via Directus.

### Errors Shown

insert into <tablename> (<columnname>) output inserted.[id] values (@p0) - The target table '<tablename>' of the DML statement cannot have any enabled triggers if the statement contains an OUTPUT clause without INTO clause. 

### What version of Directus are you using?

v9.0.0

### What version of Node.js are you using?

v16.10.0

### What database are you using?

MSSQL

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying Directus?

running locally

**Comment by wc-matteo** [comment]
yes, I hit this as well. I had to remove the triggers. But before directus implemented the output clause, the pk returned after inserts was wrong for values generated outside directus (e.g. guids generated by the db). I'm curious how this is gonna get solved... the stack overflow answer suggests:

### What about INSERTs
Do what Entity Framework does:

```sql
SET NOCOUNT ON;

DECLARE @generated_keys table([CustomerID] int)

INSERT Customers (FirstName, LastName)
OUTPUT inserted.[CustomerID] INTO @generated_keys
VALUES ('Steve', 'Brown')

SELECT t.[CustomerID], t.[CustomerGuid], t.[RowVersion], t.[CreatedDate]
FROM @generated_keys AS g
   INNER JOIN Customers AS t
   ON g.[CustomerGUID] = t.[CustomerGUID]
WHERE @@ROWCOUNT > 0
```

Again, they use a SELECT statement to read the row, rather than placing any trust in the OUTPUT clause.

**Comment by paulboudewijn** [comment]
> I had to remove the triggers.

Hi @wc-matteo ,

I've removed the triggers aswell, for me a workaround was rewriting them to stored procedures and executing those through API hooks:

```
module.exports = function registerHook({ action }) {
	action('<collection name>.items.create', (input, { database }) => {
		database.raw('exec [dbo].[<storedproc name>]').then(() => {
			console.log("stored procedure executed");
		})
		.catch((e) => {
			console.log("error executing stored procedure. see error log for details.");
			console.error(e);
		});
	});
};
```

**Comment by rijkvanzanten** [comment]
> I'm curious how this is gonna get solved... the stack overflow answer suggests:

Same here! Honestly a little surprised at how peculiar MS SQL can be for some of these subjects compared to the other supported vendors :| 

Like @wc-matteo mentioned, we used to rely on SELECT to retrieve the last inserted item, but that comes with a range of problems of its own..

**Comment by wc-matteo** [comment]
> peculiar

yes, mssql sucks .-.
